drivers/FileHandle.h

```diff
@@ -25,62 +25,85 @@ namespace mbed {
 /** \addtogroup drivers */
 /** @{*/
 
-/** An OO equivalent of the internal FILEHANDLE variable
+
+/** Class FileHandle
+ *
+ *  An OO equivalent of the internal FILEHANDLE variable
  *  and associated _sys_* functions.
  *
- * FileHandle is an abstract class, needing at least sys_write and
+ *  FileHandle is an abstract class, needing at least sys_write and
  *  sys_read to be implmented for a simple interactive device.
  *
- * No one ever directly tals to/instanciates a FileHandle - it gets
- *  created by FileSystem, and wrapped up by stdio.
- *
- * @Note Synchronization level: Set by subclass
+ *  @note to create a file, @see File
+ *  @note Synchronization level: Set by subclass
  */
 class FileHandle {
-
 public:
-    MBED_DEPRECATED_SINCE("mbed-os-5.4",
-        "The mbed 2 filesystem classes have been superseeded by the FileSystem api, "
-        "Replaced by File")
-    FileHandle() {}
+    virtual ~FileHandle() {}
 
-    /** Write the contents of a buffer to the file
+    /** Read the contents of a file into a buffer
      *
-     *  @param buffer the buffer to write from
-     *  @param length the number of characters to write
+     *  @param buffer   The buffer to read in to
+     *  @param size     The number of bytes to read
+     *  @return         The number of bytes read, 0 at end of file, negative error on failure
+     */
+    virtual ssize_t read(void *buffer, size_t len) = 0;
+
+    /** Write the contents of a buffer to a file
      *
-     *  @returns
-     *  The number of characters written (possibly 0) on success, -1 on error.
+     *  @param buffer   The buffer to write from
+     *  @param size     The number of bytes to write 
+     *  @return         The number of bytes written, negative error on failure
      */
-    virtual ssize_t write(const void* buffer, size_t length) = 0;
+    virtual ssize_t write(const void *buffer, size_t len) = 0;
 
-    /** Close the file
+    /** Close a file
      *
-     *  @returns
-     *  Zero on success, -1 on error.
+     *  @return         0 on success, negative error code on failure
      */
     virtual int close() = 0;
 
-    /** Function read
-     *  Reads the contents of the file into a buffer
+    /** Flush any buffers associated with the file
      *
-     *  @param buffer the buffer to read in to
-     *  @param length the number of characters to read
-     *
-     *  @returns
-     *  The number of characters read (zero at end of file) on success, -1 on error.
+     *  @return         0 on success, negative error code on failure
      */
-    virtual ssize_t read(void* buffer, size_t length) = 0;
+    virtual int sync() = 0;
```

Comment:
Not really buying this argument. Non-standard naming for standard things is confusing. This has driven me nuts in the past - eg Nanostack is still suffering from the fact that someone called its socket `write` function `send`. I can't fix that breakage to introduce the standard send that takes flags (because C, so no overloading).

If this API wasn't so obviously POSIX, the deviation from POSIX naming wouldn't really be an issue. The fact that it is otherwise exactly correlates to POSIX is what makes it look like the name difference might have some meaning. You should at least document that this is equivalent to POSIX fsync().

(Wasn't there a whole lot of discussion about introducing standard POSIX file system concepts recently too? That's why it's surprising me to see moves away from POSIX. Or is there an intent that we provide totally standard C fsync(), and people use it?)

How's about retaining fsync(), but not deprecating it? Appeasing both "neat names" and "POSIX names" advocates. Not sure which one calls which in that case.